gui/checkthread.cpp

```diff
@@ -509,15 +509,19 @@ QString CheckThread::getAddonFilePath(const QString &dataDir, const QString &add
 
     if (!dataDir.isEmpty()) {
         foreach (const QString p, paths) {
-            if (QFileInfo(dataDir + p + addonFile).exists())
-                return dataDir + p + addonFile;
+            QString file(dataDir);
```

Comment:
I prefer to append strings in this use case instead of combining them by the plus operator.

Reply:
why?

Reply:
Can [the following information](https://doc.qt.io/qt-5/qstring.html#append "Description for QString::append()") matter also for you?

```
…
The append() function is typically very fast (constant time), because QString preallocates extra space at the end of the string data so it can grow without reallocating the entire string each time.
…
```

Reply:
>so it's only about readability

Indeed, performance doesn't matter here at all.

Reply:
Do you find [information from an other article](https://abseil.io/tips/3 "Tip of the week #3: String concatenation and operator+ vs. StrCat()") interesting?

Reply:
I think your proposed code is not that readable. The `+` improves readability. The `abs1::StrCat` also seems more readable.

Reply:
Will the different view on source code readability result in the consequence that you are going to integrate only your own solution variant in this case?

Reply:
yes unless you see some alternative that is more readable.

Reply:
Which approach will be picked up finally?
* absl::StrCat
* append
* `+` (plus)

Reply:
not `absl::StrCat` because we don't use that. I just said it's more readable in case you can find some similar QString pattern that you like better.

not `append` the way you wrote it

`+` this is my favourite

Reply:
* I am curious if functionality like absl::StrCat will ever appear in other software libraries.
* You can take the suboptimal favourite in the meantime.